src/services/EdFi.DataManagementService.Api/Configuration/AppSettings.cs

```diff
@@ -3,12 +3,23 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
-namespace EdFi.DataManagementService.Api.Configuration
+using Microsoft.Extensions.Options;
+
+namespace EdFi.DataManagementService.Api.Configuration;
+
+public class AppSettings
+{
+    public int BeginAllowedSchoolYear { get; set; }
+    public int EndAllowedSchoolYear { get; set; }
+    public required string AuthenticationService { get; set; }
+}
+
+public class AppSettingsValidator : IValidateOptions<AppSettings>
 {
-    public class AppSettings
+    public ValidateOptionsResult Validate(string? name, AppSettings options)
     {
-        public int BeginAllowedSchoolYear { get; set; }
-        public int EndAllowedSchoolYear { get; set; }
-        public required string AuthenticationService { get; set; }
+        return options.AuthenticationService == null
+            ? ValidateOptionsResult.Fail("Missing required AppSettings value: AuthenticationService")
```

Comment:
Interesting, in the log this comes through as an unhandled exception. Obviously we _did_ handle it - resulting in a log message and a 500. It is unsettling to see a full stack trace and "An uncaught error has occurred", making it look like we forgot to put in some important error handling.